Resources/Private/JavaScript/Host/Containers/SecondaryToolbar/DimensionSwitcher/index.js

```diff
@@ -1,48 +1,108 @@
+    icon: PropTypes.string.isRequired,
+    dimensionLabel: PropTypes.string.isRequired,
+    presetLabel: PropTypes.string.isRequired,
+    dimensionName: PropTypes.string.isRequired
+};
+
+const DimensionSelector = props => {
+    const {icon, dimensionLabel, presets, dimensionName, activePreset, onSelect} = props;
+
+    const presetOptions = presets.map((presetConfiguration, presetName) => ({
+        label: presetConfiguration.get('label'),
+        value: presetName
+    })).toArray();
 
-            ToDo: Render the select for all dimension items
+    const onPresetSelect = presetName => {
+        onSelect(dimensionName, presetName);
+    };
+
+    return (
+        <li key={dimensionName} className={style.dimensionCategory}>
+            <Icon icon={icon} padded="right" className={style.dimensionCategory__icon}/>
+            <I18n id={dimensionLabel} />
+            <br/>
+            <SelectBox options={presetOptions} onSelect={onPresetSelect} value={activePreset} />
         </li>
     );
 };
-DimensionCategory.propTypes = {
-    data: PropTypes.object.isRequired,
-    key: PropTypes.number
+DimensionSelector.propTypes = {
+    icon: PropTypes.string.isRequired,
+    dimensionLabel: PropTypes.string.isRequired,
+    presets: PropTypes.object.isRequired,
+    activePreset: PropTypes.string.isRequired,
+    dimensionName: PropTypes.string.isRequired,
+    onSelect: PropTypes.func.isRequired
 };
 
-@connect()
+@connect($transform({
+    contentDimensions: $get('cr.contentDimensions.byName'),
+    activePresets: selectors.CR.ContentDimensions.activePresets
+}), {
+    selectPreset: actions.CR.ContentDimensions.selectPreset
+})
 export default class DimensionSwitcher extends Component {
     static propTypes = {
-        dimensions: PropTypes.array.isRequired
+        contentDimensions: PropTypes.object.isRequired,
+        activePresets: PropTypes.object.isRequired
     };
 
     static defaultProps = {
-        dimensions: []
+        contentDimensions: new Map(),
+        activePresets: new Map()
     };
 
+    shouldComponentUpdate(...args) {
+        return shallowCompare(this, ...args);
+    }
+
     render() {
-        const {dimensions} = this.props;
+        const {contentDimensions, activePresets, selectPreset} = this.props;
 
         return (
             <DropDown className={style.dropDown}>
                 <DropDown.Header className={style.dropDown__btn}>
-                    <Icon className={style.dropDown__btnIcon} icon="globe"/>
-                    ToDo: Current dimension name
+                    {contentDimensions.map((dimensionConfiguration, dimensionName) =>
+                        <SelectedPreset
+                            dimensionName={dimensionName}
+                            icon={dimensionConfiguration.get('icon')}
+                            dimensionLabel={dimensionConfiguration.get('label')}
+                            presetLabel={activePresets.get(dimensionName).get('label')} />
```

Comment:
here I'd rather use Plow.js, like `$get([dimensionName, 'label'], activePresets)` -- so to not couple the GET to Immutable.JS.